### ISSUE TITLE
[Feature Request] Setting namespace and task queue during replay

### ISSUE THREAD

**Issue by cretz** [temporalio/features#114]
### Is your feature request related to a problem? Please describe.

Some SDKs populate these values with garbage or test data during replay which affects people's workflow code

### Describe the solution you'd like

We should accept an (optional) namespace in replay options and take the task queue out of the start event. Python will already do this on the first version.

**Comment by josh-berry** [comment]
Is this still a thing? What fixes do we need in which SDKs?

**Comment by cretz** [comment]
Yes, this needs to be assigned at least to confirm behavior. We probably need to confirm task queue is coming from history in Core/Java, and namespace is settable in Go/TS/Java. But I don't think anyone is clamoring for this.
